app/controllers/participants_controller.rb

```diff
@@ -1,9 +1,15 @@
 class ParticipantsController < ApplicationController
-  
+
   before_action :set_participant, only: [:show, :edit, :update, :destroy]
 
   # GET /participants
   # GET /participants.json
+
+  def assign_santas
+    @assign_santas = Participant.all.shuffle.zip(Participant.all)
```

Comment:
LOL! You are very correct! 
I just got the feature working, and finally showing the array in the view and realised that my mother (a test participant in the db) has herself for Christmas :( 
How might I go about making sure this doesn’t happen?
Also how did you find this? Was it from my pull request? It’s my first time practicing with pull requests etc.
Thanks for reaching out to me! :D
-- 
Owen Schebella
Sent with Airmail

On 14 March 2016 at 4:19:20 pm, MJ (notifications@github.com) wrote:

In app/controllers/participants_controller.rb:

>    before_action :set_participant, only: [:show, :edit, :update, :destroy]
>   
>    # GET /participants
>    # GET /participants.json
> +
> +  def assign_santas
> +    @assign_santas = Participant.all.shuffle.zip(Participant.all)
If I understand the intention of this algorithm correctly, it might need some tweaking as there is a possibility that Participants will be paired with themselves (and that person will be sad for christmas  ) e.g.,

![Uploading serviceseeking_—__Users_markagbuya_code_serviceseeking_—_ruby_—_238×55.png…]()

—
Reply to this email directly or view it on GitHub.